transfer-items.lic

```diff
@@ -2,41 +2,32 @@
   Documentation: https://elanthipedia.play.net/Lich_script_repository#transfer-items
 =end
 
+custom_require.call %w[common common-items]
+
 class ItemTransfer
+  include DRC
+  include DRCI
+
   def initialize
     arg_definitions = [
       [
-        { name: 'source', regex: /^[A-z\s\-]+$/i, variable: true, description: 'Source container' },
-        { name: 'destination', regex: /^[A-z\s\-]+$/i, variable: true, description: 'Destination container' }
+        { name: 'source', regex: /^[A-z\.\s\-]+$/i, variable: true, description: 'Source container' },
+        { name: 'destination', regex: /^[A-z\.\s\-]+$/i, variable: true, description: 'Destination container' }
       ]
     ]
     args = parse_args(arg_definitions)
-    @source = args.source
-    @destination = args.destination
-
-    transfer_items
+    transfer_items(args.source, args.destination)
   end
 
-  def transfer_items
-    fput("look in my #{@source}")
-
-    while line = get
-      if line =~ /In the .*#{@source} you see/
-        items = line
-        break
-      end
-    end
-
-    items
-      .sub(' and ', ',')
-      .split(',')
-      .map { |full_name| full_name.split(' ') }
-      .map { |parts| parts.last.sub('.', '') }
+  def transfer_items(source, destination)
+    DRCI.get_item_list(source)
+      .map { |full_name| full_name.split(' ').last }
       .each do |item|
-        fput("get #{item} from my #{@source}")
-        fput("put #{item} in my #{@destination}")
+        fput("get #{item} from my #{source}")
```

Comment:
💡 Alternatively, these could be changed to `bputs` (and keep `common` above)

Reply:
Good point, I'll submit a revision using `bput`